app/src/main/java/com/iic/shopingo/api/Server.java

```
@@ -0,0 +1,101 @@
+  public <T> T get(String path, Class<T> responseClass) throws IOException {
+    return get(path, responseClass, new HashMap<String, Object>());
+  }
+
+  public <T> T get(String path, Class<T> responseClass, Map<String, Object> params) throws IOException {
+    Uri.Builder requestUriBuilder = BASE_URI.buildUpon().encodedPath(path);
+    for (String key : params.keySet()) {
+      requestUriBuilder.appendQueryParameter(key, params.get(key).toString());
+    }
+    HttpGet request = new HttpGet(requestUriBuilder.build().toString());
+    addAuthData(request);
+    return executeRequest(request, responseClass);
+  }
+
+  public <T> T post(String path, Class<T> responseClass) throws IOException {
+    return post(path, responseClass, new HashMap<String, Object>());
+  }
+
+  public <T> T post(String path, Class<T> responseClass, Map<String, Object> params) throws IOException {
+    Uri.Builder requsetUriBuilder = BASE_URI.buildUpon().encodedPath(path);
+    HttpPost request = new HttpPost(requsetUriBuilder.build().toString());
+    request.setHeader("Content-Type", "application/json");
+    request.setHeader("Accept", "application/json");
+    addAuthData(request);
+    JSONObject obj = new JSONObject(params);
+    request.setEntity(new StringEntity(obj.toString()));
+    return executeRequest(request, responseClass);
+  }
+
+  private void addAuthData(HttpUriRequest request) {
+    if (authToken != null) {
+      request.addHeader(HEADER_AUTH_KEY, authToken);
+    }
+  }
+
+  private <T> T executeRequest(HttpUriRequest request, Class<T> responseClass) throws IOException {
+    InputStream content = null;
+    try {
+      HttpResponse response = client.execute(request);
+      content = response.getEntity().getContent();
+      Reader reader = new InputStreamReader(content);
+      Gson gson = new GsonBuilder().create();
+      T result = gson.fromJson(reader, responseClass);
+      content.close();
+      return result;
+    } catch (Exception e) {
+      throw e;
```

If you just have a try/finally it eats all the exceptions.